### PR TITLE
Add CupertinoTextFormField widget

### LINKED ISSUE #26745: Create CupertinoTextFormField
`TextFormField` supports parameter `validator`
`CupertinoTextField` now exists

so... adding `CupertinoTextFormField` supporting `validator` parameter would be great

### PR DESCRIPTION
**Pull request by mattarau**
## Description

This is an adaptation of the already existent `TextFormField` Material widget, which is a `FormField` wrapper of the `TextField`, so that the existent `CupertinoTextField` can also easily be used in forms. 

## Related Issues

Fixes https://github.com/flutter/flutter/issues/26745

## Tests

I have kept all the relevant tests from `TextFormField` and added the following:
- Passes padding to underlying CupertinoTextField
- Passes scrollPadding to underlying CupertinoTextField
- Passes keyboardAppearance to underlying CupertinoTextField
- Passes placeholder to underlying CupertinoTextField
- Passes decoration to underlying CupertinoTextField

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/text_form_field.dart [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    Radius cursorRadius,
+    Color cursorColor,
+    Brightness keyboardAppearance,
+    EdgeInsets scrollPadding = const EdgeInsets.all(20.0),
+    DragStartBehavior dragStartBehavior = DragStartBehavior.start,
+    bool enableInteractiveSelection = true,
+    GestureTapCallback onTap,
+    ScrollController scrollController,
+    ScrollPhysics scrollPhysics,
+
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/text_form_field.dart:128]: nit: extra space

File: packages/flutter/lib/src/cupertino/text_form_field.dart [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+         !expands || (maxLines == null && minLines == null),
+         'minLines and maxLines must be null when expands is true.',
+       ),
+       assert(maxLength == null || maxLength > 0),
+       assert(enableInteractiveSelection != null),
+       assert(clearButtonMode != null),
+       assert(prefixMode != null),
+       assert(suffixMode != null),
+       super(
+    key: key,
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/text_form_field.dart:156]: nit: the indent is weird here.

**Review by xster — COMMENTED** [review]
Great. Thanks for your contribution. This LGTM. Just some nits.

**Comment by Piinks** [comment]
@mattarau would you like to return to this pull request?

**Comment by mattarau** [comment]
Hey @Piinks / @xster 

I'll get to this soon. Probably next week.

Cheers!

**Comment by mattarau** [comment]
Hey @xster I have removed the empty line, but regarding the indentation I think this should be taken care as a separated issue (if necessary) as it is just how the formatter does it.

You can see it not just in the Material version of this Widget (where I got the source from):
https://github.com/flutter/flutter/blob/4d7525f05c05a6df0b29396bc9eb78c3bf1e9f89/packages/flutter/lib/src/material/text_form_field.dart#L205-L206,

But also on other files, like this one: https://github.com/flutter/flutter/blob/8c3b826ebd550008a23b4e2f50b249facff3004b/dev/tools/gen_keycodes/lib/key_data.dart#L44-L45

Let me know if there's anything else missing. ;)

**Comment by xster** [comment]
Could you rebase your branch to the tip of the master branch? I suspect the CI issues are from the old branch state.

**Comment by Piinks** [comment]
This PR has not been updated in a while so I am going to close it for now. If you would like to come back and work on it, feel free to re-open! Thanks!

**Pull request closed without merging**